### PR TITLE
Slack/api metrics

### PR DESCRIPTION
**Pull request by aloknerurkar**
Adding counter for error codes that are returned eg:
```
# TYPE bee_api_reponse_error_codes counter
bee_api_reponse_error_codes{code="404",method="GET"} 3
```

### CODE REVIEW

File: pkg/api/metrics.go [hunk 1]
```diff
@@ -38,6 +40,15 @@ func newMetrics() metrics {
 			Help:      "Histogram of API response durations.",
 			Buckets:   []float64{0.01, 0.1, 0.25, 0.5, 1, 2.5, 5, 10},
 		}),
+		ErrorPageViews: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: m.Namespace,
+				Subsystem: subsystem,
+				Name:      "reponse_error_codes",
+				Help:      "How many 5XX HTTP errors were returned",
```
Comment (esadakar) [inline pkg/api/metrics.go:48]: is it 4xx or 5xx

**Review by esadakar — APPROVED** [review]

File: pkg/api/metrics.go [hunk 2]
```diff
@@ -38,6 +40,15 @@ func newMetrics() metrics {
 			Help:      "Histogram of API response durations.",
 			Buckets:   []float64{0.01, 0.1, 0.25, 0.5, 1, 2.5, 5, 10},
 		}),
+		ErrorPageViews: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: m.Namespace,
+				Subsystem: subsystem,
+				Name:      "reponse_error_codes",
```
Comment (acud) [inline pkg/api/metrics.go:47]: typo in `reponse`

File: pkg/api/metrics.go [hunk 3]
```diff
@@ -19,6 +20,7 @@ type metrics struct {
 	RequestCount     prometheus.Counter
 	ResponseDuration prometheus.Histogram
 	PingRequestCount prometheus.Counter
+	ErrorPageViews   *prometheus.CounterVec
```
Comment (acud) [inline pkg/api/metrics.go:23]: I would consider having a breakdown of all status code groups - 1xx 2xx 3xx 4xx 5xx

Comment (aloknerurkar) [reply pkg/api/metrics.go]: Just mentioning our discussion here for record.

We have decided to do the grouping on graphana side instead. It seems to be better to not have the extra lookups on the client-side.

**Review by acud — APPROVED** [review]

**Review by acud — APPROVED** [review]

**Pull request merged**